pkgs/development/compilers/emscripten/default.nix

```diff
@@ -2,31 +2,39 @@
 , jre, binaryen
 , llvmPackages
 , symlinkJoin, makeWrapper, substituteAll
-, mkYarnModules
+, buildNpmPackage
 }:
 
 stdenv.mkDerivation rec {
   pname = "emscripten";
-  version = "3.1.17";
+  version = "3.1.24";
 
   llvmEnv = symlinkJoin {
     name = "emscripten-llvm-${version}";
     paths = with llvmPackages; [ clang-unwrapped clang-unwrapped.lib lld llvm ];
   };
 
-  nodeModules = mkYarnModules {
+  nodeModules = buildNpmPackage {
     name = "emscripten-node-modules-${version}";
-    inherit pname version;
-    # it is vitally important the the package.json has name and version fields
-    packageJSON = ./package.json;
-    yarnLock = ./yarn.lock;
-    yarnNix = ./yarn.nix;
+    inherit pname version src;
+
+    npmDepsHash = "sha256-ejuHR2BpAUStWjuvQuGE6ko4byF4GBl6FJBshxlknQk=";
+
+    # Do not build.
+    buildPhase = ''
+      echo no build
```

Comment:
I guess `dontNpmBuild = true; dontBuild = true;` would work here?

Reply:
dontBuild is it probaby.